src/HTSPConnection.cpp

```diff
@@ -107,23 +107,24 @@ CHTSPConnection::~CHTSPConnection()
  * Info
  */
 
-CStdString CHTSPConnection::GetWebURL ( const char *fmt, ... )
+std::string CHTSPConnection::GetWebURL ( const char *fmt, ... )
 {
   va_list va;
-  CStdString auth, url;
   tvheadend::Settings settings = tvh->GetSettings();
 
-  auth = settings.strUsername;
-  if (auth != "" && settings.strPassword != "")
+  // Generate the authentication string (user:pass@)
+  std::string auth = settings.strUsername;
+  if (!auth.empty() && !settings.strPassword.empty())
```

Comment:
I'd prefer <code>if (!(auth.empty() || settings.strPassword.empty()))</code>. Reads IMO more "natural", but leave it as is if you don't feel the same.